lib/kino/explorer.ex

```diff
@@ -0,0 +1,135 @@
+defmodule Kino.Explorer do
+  @moduledoc """
+  A widget for interactively viewing `Explorer.DataFrame`.
+  ## Examples
+      df = Explorer.Datasets.fossil_fuels()
+      Kino.Explorer.new(df)
+  """
+
+  @behaviour Kino.Table
+
+  @type t :: Kino.JS.Live.t()
+
+  @doc """
+  Starts a widget process representing the given data frame.
+  """
+  @spec new(Explorer.DataFrame.t()) :: t()
+  def new(df) do
+    Kino.Table.new(__MODULE__, {df})
+  end
+
+  @doc false
+  def static(%Explorer.DataFrame{} = df) do
+    Kino.Table.new(__MODULE__, {df})
+  end
+
+  @impl true
+  def init({df}) do
+    total_rows = Explorer.DataFrame.n_rows(df)
+    dtypes = Explorer.DataFrame.dtypes(df)
+    sample_data = df |> Explorer.DataFrame.head(1) |> Explorer.DataFrame.to_columns()
+    summaries = summary(df)
+
+    columns =
+      Enum.map(dtypes, fn {name, dtype} ->
+        %{
+          key: name,
+          label: to_string(name),
+          type: type_of(dtype, sample_data[name]),
+          summary: summaries[String.to_atom(name)]
+        }
+      end)
+
+    info = %{name: "DataFrame", features: [:pagination, :sorting]}
+
+    {:ok, info, %{df: df, total_rows: total_rows, columns: columns}}
+  end
+
+  @impl true
+  def get_data(rows_spec, state) do
+    records = get_records(state.df, rows_spec)
+    rows = Enum.map(records, &record_to_row/1)
+    {:ok, %{columns: state.columns, rows: rows, total_rows: state.total_rows}, state}
+  end
+
+  defp get_records(df, rows_spec) do
+    df =
+      if order_by = rows_spec[:order_by] do
+        Explorer.DataFrame.arrange_with(df, &[{rows_spec.order, &1[order_by]}])
+      else
+        df
+      end
+
+    df = Explorer.DataFrame.slice(df, rows_spec.offset, rows_spec.limit)
+
+    {cols, lists} = df |> Explorer.DataFrame.to_columns() |> Enum.unzip()
+    col_names = Enum.map(cols, &to_string/1)
+
+    lists
+    |> Enum.zip()
+    |> Enum.map(fn row ->
+      Enum.zip(col_names, Tuple.to_list(row))
+    end)
+  end
+
+  defp record_to_row(record) do
+    fields = Map.new(record, fn {col_name, value} -> {col_name, to_string(value)} end)
+    %{fields: fields}
+  end
+
+  defp summary(df) do
+    describe = describe(df)
+    df_series = Explorer.DataFrame.to_series(df)
+
+    for {column, [mean, min, max]} <- describe,
+        series = Map.get(df_series, column),
+        summary_type = summary_type(series),
+        nulls = Explorer.Series.nil_count(series) |> to_string(),
+        label = String.to_atom(column) do
```

Comment:
We can remove `to_atom` here and in the other place too. You can do `into: %{}` so access still works :)